### DISCUSSION TITLE
How to add a custom tool to Conversational Retrieval QA

### DISCUSSION THREAD

**Discussion by davidkobielski** [FlowiseAI/Flowise#702] [Q&A] [answered]
Hello!
I'm creating my first chatbot using flowise and I have a problem...
I want chatbot to response with provided data by me, but also I want to add custom tool. Here is my actual flow:
![image](https://github.com/FlowiseAI/Flowise/assets/42317458/0205cceb-64ab-44c1-b2a6-b065ef563024)

How can I connect Conversational Retrieval QA Chain with custom tool? I know it's possible to connect a chain to agent using Chain Tool, but when I did this, my chatbot didn't follow all the instructions... I mean, it was working, but didn't care about my system message. How can I create a bot, that will send a response based on custom data, has a custom tool, follow system instructions and don't response on questions not regarding company?

![image](https://github.com/FlowiseAI/Flowise/assets/42317458/bd20636e-c602-440b-9182-3f5199b5f6f5)

**Comment by HenryHengZJ** [comment, marked as answer]
We have a new conversational retrieval agent coming up in next release that allow you to use Vector Store Retriever as tool

**Reply by davidkobielski** [reply]
Thank you Henry for the response! Great to hear that! It will solve a lot of problems. Has the date of the new release been established?

**Reply by HenryHengZJ** [reply]
released now!

**Reply by davidkobielski** [reply]
That makes life a lot easier :) Thank you! Still have problem with not following my instructions, but I think it's because model GPT-3,5 it works bad with such a case... The bot should response only with data about the company, but when I ask him who is Cristiano Ronaldo it reponse his history of life. I think, I have to get GPT-4 API to better follow my instructions. Or maybe you have an idea as an expert? Have a nice day!
